modules/rational-defaults.el

```diff
@@ -11,14 +11,44 @@
 
 ;;; Code:
 
+;; Defining variable for XDG directories specifications
+(defvar xdg-data-home-directory
+  (if (getenv "XDG_DATA_HOME")
+    (getenv "XDG_DATA_HOME")
+    (expand-file-name ".local/share/")))
+
+(defvar xdg-cache-home-directory
+  (if (getenv "XDG_DATA_HOME")
+    (getenv "XDG_DATA_HOME")
+    (expand-file-name ".cache/")))
+
+;; Keep emacs folder clean
+
+;; Backup file ( init.el~ )
+(customize-set-variable backup-directory-alist `(("." . ,(expand-file-name "emacs/backups/" xdg-data-home-directory))))
+;; Emacs autosave files ( #init.el# )
+;; The autosave feature doesn't create the directory
+(make-directory (expand-file-name "emacs/auto-save/" xdg-cache-home-directory) t)
+(customize-set-variable auto-save-file-name-transforms
+    `((".*" ,(expand-file-name "emacs/auto-save/" xdg-cache-home-directory) t)))
+
+;; eshell history, lastdir and aliases
+(customize-set-variable eshell-directory-name 
+  (expand-file-name "emacs/eshell" xdg-data-home-directory))
+(customize-set-variable eshell-aliases-file 
+  (expand-file-name "emacs/eshell/aliases" xdg-data-home-directory))
+;; bookmark default file
+(customize-set-variable bookmark-default-file 
```

Comment:
There is currently a draft PR to move other files (bookmarks, saveplace, tramp, etc) to the `rational-config-var-directory`. This probably belongs there as well.